unstructured_ingest/v2/processes/connectors/vectara.py

```diff
@@ -0,0 +1,312 @@
+            response_json = response.json()
+
+        request_time = datetime.now().timestamp()
+        self._jwt_token_expires_ts = request_time + response_json.get("expires_in")
+
+        return response_json.get("access_token")
+
+    @DestinationConnectionError.wrap
+    async def _check_connection_and_corpora(self) -> None:
+        """
+        Check the connection for Vectara and validate corpus exists.
+        - If more than one corpus with the same name exists - then return a message
+        - If exactly one corpus exists with this name - use it.
+        - If does not exist - create it.
+        """
+        # Get token if not already set
+        await self.jwt_token
+
+        _, list_corpora_response = await self._request(
+            http_method="GET",
+            endpoint="corpora",
+        )
+
+        if self.connection_config.corpus_name:
+            possible_corpora_keys_names_map = {
+                corpus.get("key"): corpus.get("name")
+                for corpus in list_corpora_response.get("corpora")
+                if corpus.get("name") == self.connection_config.corpus_name
+            }
+
+            if len(possible_corpora_keys_names_map) > 1:
+                raise ValueError(
+                    f"Multiple Corpus exist with name {self.connection_config.corpus_name} in dest."
+                )
+            if len(possible_corpora_keys_names_map) == 1:
+                if not self.connection_config.corpus_key:
+                    self.connection_config.corpus_key = list(
+                        possible_corpora_keys_names_map.keys()
+                    )[0]
+                elif (
+                    self.connection_config.corpus_key
+                    != list(possible_corpora_keys_names_map.keys())[0]
+                ):
+                    raise ValueError("Corpus key does not match provided corpus name.")
+            else:
+                raise ValueError(
+                    f"No Corpora exist with name {self.connection_config.corpus_name} in dest."
+                )
+
+    @requires_dependencies(["httpx"], extras="vectara")
+    async def _request(
+        self,
+        endpoint: str,
+        http_method: str = "POST",
+        params: Mapping[str, Any] = None,
+        data: Mapping[str, Any] = None,
+    ) -> tuple[bool, dict]:
+        import httpx
+
+        url = f"{BASE_URL}/{endpoint}"
+
+        headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Authorization": f"Bearer {await self.jwt_token}",
+            "X-source": "unstructured",
+        }
+
+        async with httpx.AsyncClient() as client:
+            response = await client.request(
+                method=http_method, url=url, headers=headers, params=params, json=data
+            )
+            response.raise_for_status()
+            return response.is_success, response.json()
+
+    async def _delete_doc(self, doc_id: str) -> None:
```

Comment:
This function here uses `self._request` , so also should return `tuple[bool, dict]`